### PR TITLE
allow empty headers

### LINKED ISSUE #69: Error when no headers in request to API gateway
I've got a normal setup, based closely on the example in magnum-cli and terraform docs on AWS lambda.  Using the AWS api gateway test button throws this error:

```yml
# loosely translated from the API gateway log
Endpoint response body before transformations:
  errorMessage:
    message: 'NoneType' object has no attribute 'get'
    errorType: AttributeError
    stackTrace: >
      File "/var/task/mangum/adapter.py", line 40, in __call__
        raise exc
      File "/var/task/mangum/adapter.py", line 35, in __call__
        response = self.handler(event, context)
      File "/var/task/mangum/adapter.py", line 50, in handler
        response = handle_http(self.app, event, context)
      File "/var/task/mangum/protocols/http.py", line 58, in handle_http
        server, client = get_server_and_client(event)
      File "/var/task/mangum/utils.py", line 22, in get_server_and_client
        server_addr = event["headers"].get("Host", None)
```

Adding `Accept:application/json` to the textbox for headers resolves the issue, and I get my expected `{ greeting: "hello" }` JSON back.

### PR DESCRIPTION
**Pull request by SKalt**
resolves #69.

### CODE REVIEW

File: mangum/adapter.py [hunk 1]
```diff
@@ -57,8 +57,9 @@ def handler(self, event: dict, context: dict) -> dict:
 
     def handle_http(self, event: dict, context: dict) -> dict:
         server, client = get_server_and_client(event)
-        headers = [
-            [k.lower().encode(), v.encode()] for k, v in event["headers"].items()
+        headers = event.get("headers", dict())
```
Comment (erm) [inline mangum/adapter.py]: I'd prefer to use the dict literal syntax here, e.g. `event.get("headers", {})`

File: mangum/adapter.py [hunk 2]
```diff
@@ -112,15 +113,15 @@ def handle_ws(self, event: dict, context: dict) -> dict:
             # The initial connect event. Parse and store the scope for the connection
             # in DynamoDB to be retrieved in subsequent message events for this request.
             server, client = get_server_and_client(event)
-            headers = event["headers"]
+            headers = event.get("headers", dict())
```
Comment (erm) [inline mangum/adapter.py]: We want to duplicate the header definition in the HTTP handler for WebSockets as well.

**Comment by erm** [comment]
@SKalt I've added a couple minor comments, but otherwise looks fine - thanks!

File: mangum/adapter.py (hunk 1 above)
Comment (SKalt) [reply mangum/adapter.py]: changes pushed.

File: mangum/adapter.py (hunk 2 above)
Comment (SKalt) [reply mangum/adapter.py]: I swapped this line for a dict literal as well.  Was that the change you were looking for?

**Comment by erm** [comment]
Thanks! :)

**Pull request merged**